ksqldb-rest-app/src/test/java/io/confluent/ksql/api/auth/AuthenticationPluginHandlerTest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.auth;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.regex.Pattern;
+import org.junit.Test;
+
+public class AuthenticationPluginHandlerTest {
+
+  @Test
+  public void shouldBuildRegexThatRespectsSkipPaths() {
+    // Given:
+    final List<String> configs = ImmutableList.of("/heartbeat", "/lag");
+
+    // When:
+    final Pattern skips = AuthenticationPluginHandler.getAuthorizationSkipPaths(configs);
+
+    // Then:
+    assertThat(skips.matcher("/heartbeat").matches(), is(true));
```

Comment:
Why not also verify for `/lags`? 
Could you also add a negative case, like `/shouldnotbefound`?

Reply:
> Why not also verify for /lags?

that doesn't add any coverage, I'm ensuring that what is added in the config is able to skip

> Could you also add a negative case, like /shouldnotbefound?

Sure I'll add a negative test

Reply:
adding `/lag` in the config list makes sure that the regex works when you have multiple components - I'm not sure why checking that it works for `/lag` adds any coverage beyond checking for `/heartbeat` (basically it uses the same exact code path to check for `/lag` as it does to check for `/heartbeat`)

Reply:
I suppose I can see your point :) I'll add it in.

Reply:
https://github.com/confluentinc/ksql/pull/9229